ANMN/burst_averaged_product/burst_average.py

```diff
@@ -0,0 +1,397 @@
+    output_netcdf_obj.geospatial_lon_max = input_netcdf_obj['LONGITUDE'][:].max()
+
+    output_netcdf_obj.geospatial_vertical_min = min(burst_vars['DEPTH']['var_mean'])
+    output_netcdf_obj.geospatial_vertical_max = max(burst_vars['DEPTH']['var_mean'])
+
+    output_netcdf_obj.time_coverage_start = num2date(time_burst_vals, input_netcdf_obj['TIME'].units, input_netcdf_obj['TIME'].calendar).min().strftime('%Y-%m-%dT%H:%M:%SZ')
+    output_netcdf_obj.time_coverage_end   = num2date(time_burst_vals, input_netcdf_obj['TIME'].units, input_netcdf_obj['TIME'].calendar).max().strftime('%Y-%m-%dT%H:%M:%SZ')
+
+    # set up dimensions
+    output_netcdf_obj.createDimension("TIME", len(time_burst_vals))
+
+    # set up variables
+    var_time = output_netcdf_obj.createVariable("TIME", "d", ("TIME",), fill_value=get_imos_parameter_info('TIME', '_FillValue'))
+    var_lat  = output_netcdf_obj.createVariable("LATITUDE", "d", fill_value=get_imos_parameter_info('LATITUDE', '_FillValue'))
+    var_lon  = output_netcdf_obj.createVariable("LONGITUDE", "d", fill_value=get_imos_parameter_info('LONGITUDE', '_FillValue'))
+
+    for var in burst_vars.keys():
+        fillvalue = get_imos_parameter_info(var, '_FillValue')
+        if fillvalue == []:
+            fillvalue = 999999.0
+
+        output_netcdf_obj.createVariable(var, "f4", ("TIME",), fill_value=fillvalue)
+        output_netcdf_obj.createVariable('%s_burst_min' % var, "f4", ("TIME",), fill_value=fillvalue)
+        output_netcdf_obj.createVariable('%s_burst_max' % var, "f4", ("TIME",), fill_value=fillvalue)
+        output_netcdf_obj.createVariable('%s_burst_sd' % var, "f4", ("TIME",), fill_value=fillvalue)
+        output_netcdf_obj.createVariable('%s_num_obs' % var, "i4", ("TIME",), fill_value=fillvalue)
+
+        # set up 'bonus' var att from original FV01 file to FV02
+        input_var_object   = input_netcdf_obj[var]
+        input_var_list_att = input_var_object.__dict__.keys()
+        var_att_disposable = ['name', 'standard_name', 'long_name', 'valid_min', \
+                              'valid_max', '_FillValue', 'ancillary_variables', \
+                              'axis', 'units', 'ChunkSize']
+        for var_att in [att for att in input_var_list_att if att not in var_att_disposable]:
+            setattr(output_netcdf_obj[var], var_att, getattr(input_netcdf_obj[var], var_att))
+            if var_att != 'comment':
+                setattr(output_netcdf_obj['%s_burst_min' % var], var_att, getattr(input_netcdf_obj[var], var_att))
+                setattr(output_netcdf_obj['%s_burst_max' % var], var_att, getattr(input_netcdf_obj[var], var_att))
+                setattr(output_netcdf_obj['%s_burst_sd' % var], var_att, getattr(input_netcdf_obj[var], var_att))
+
+        var_units = get_imos_parameter_info(var, 'units')
+        if var_units:
+            setattr(output_netcdf_obj[var], 'units', var_units)
+            setattr(output_netcdf_obj['%s_burst_min' % var], 'units', var_units)
```

Comment:
Instead of `setattr` you can assign directly to an attribute, e.g. `output_var_min.units = var_units`  (assuming you assigned `output_var_min` as per comment above)